graphql_compiler/cost_estimation/helpers.py

```diff
@@ -1,33 +1,51 @@
 # Copyright 2019-present Kensho Technologies, LLC.
-from graphql import GraphQLInt
+from typing import Union, cast
+
+from graphql import (
+    GraphQLInt,
+    GraphQLInterfaceType,
+    GraphQLList,
+    GraphQLObjectType,
+    GraphQLScalarType,
+)
 
 from ..global_utils import is_same_type
 from ..schema import GraphQLDate, GraphQLDateTime
 from ..schema.schema_info import QueryPlanningSchemaInfo
 
 
+def _get_field_type(
+    schema_info: QueryPlanningSchemaInfo, vertex_name: str, field_name: str
+) -> Union[GraphQLList, GraphQLScalarType]:
+    """Get the GraphQL type of the field on the specified vertex."""
+    return (
+        cast(
+            Union[GraphQLObjectType, GraphQLInterfaceType], schema_info.schema.get_type(vertex_name)
+        )
```

Comment:
This shouldn't be a cast, instead it should be an `isinstance` + assertion. Technically, `get_type()` can return union types, enums, etc. and while `vertex_name` _should_ only refer to interfaces and objects, it might not.